platforms/allwinner-d1/d1-config/src/lib.rs

```diff
@@ -0,0 +1,75 @@
+#![no_std]
+
+use serde::{Serialize, Deserialize};
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct PlatformConfig {
+    pub i2c: I2cConfiguration,
+    pub i2c_puppet: I2cPuppetConfiguration,
+    pub blink_service: LedBlinkService,
+}
+
+// I2C
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct I2cConfiguration {
+    #[serde(default)]
+    pub enabled: bool,
+    #[serde(default = "I2cConfiguration::default_mapping")]
+    pub mapping: Mapping,
+}
+
+impl I2cConfiguration {
+    const fn default_mapping() -> Mapping {
+        Mapping::LicheeRvTwi0
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub enum Mapping {
+    LicheeRvTwi0,
+    MangoPiTwi2,
+}
```

Comment:
all the D1s have TWI peripherals 0-3, so maybe this should be 
```suggestion
#[derive(Debug, Serialize, Deserialize)]
pub enum Twi {
    Twi0,
    // Twi1,
    Twi2,
    // Twi3,
}
```
and add a separate thing to configure which pins that twi is mapped to? but maybe that's just unnecessary complexity at this point...

Reply:
Yeah, since we don't actually support full mapping, it would be something like (pin numbers made up):

* Twi0_PB1_PB7
* Twi2_PG13_PG15

Since they can't really be decoupled.